src/Dependencies/Collections/ImmutableSegmentedDictionary.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
```

Comment:
Need to add `#nullable enable` so that it gets enabled in projects that don't have it enabled by default.

Reply:
➡️ The current code follows the patterns established in the **Microsoft.CodeAnalysis.PooledObjects.Package** package. If we want to change the way we annotate source packages, we can do so but it should be a separate pull request that includes validation to prevent subtle regressions.

Reply:
➡️ #50749